src/main/c/GCI_Phasor.c

```diff
@@ -83,9 +84,33 @@ double GCI_Phasor_getPeriod()
  chisq is calculated for comparison with other methods but is not used as there is no optimisation with this method
 */
 
-int    GCI_Phasor(float xincr, float y[], int fit_start, int fit_end,
+int GCI_Phasor(float xincr, float y[], int fit_start, int fit_end,
 							  const float *Z, float *U, float *V, float *taup, float *taum, float *tau, float *fitted, float *residuals,
 							  float *chisq)
+{
+	int nBins = (fit_end - fit_start);
+	if (nBins < 0)
+		return (PHASOR_ERR_INVALID_WINDOW);
+	float* cos = (float*)malloc((long unsigned int)nBins * sizeof(float));
+	float* sin = (float*)malloc((long unsigned int)nBins * sizeof(float));
```

Comment:
Same comment applies regarding the names `cos` and `sin`.

Also, best to cast `nBins` to `(size_t)nBins` because both `malloc()`'s parameter and the result of `sizeof()` are `size_t`.

And you don't need the cast to `(float*)` in C (and it can hide other issues). In C++ you would need it, but this code won't be compiled as C++ (and `malloc` shouldn't be used in C++ normally).